source/Halibut/ServiceModel/ServiceInvoker.cs

```diff
@@ -103,19 +209,46 @@ static MethodInfo SelectMethod(IList<MethodInfo> methods, RequestMessage request
             throw new AmbiguousMethodMatchHalibutClientException(message.ToString(), new AmbiguousMatchException(message.ToString()));
         }
 
+        async Task<object> InvokeAsyncMethod(MethodInfo method, object obj, params object[] parameters)
+        {
+            var returnType = method.ReturnType;
+
+            // If the return type is not Task or Task<T>, then shrug and explode!
+            if (!typeof(Task).IsAssignableFrom(returnType))
+            {
+                throw new Exception();
```

Comment:
Throw a more meaningful message here.

Reply:
Ah damn, I was going to change that, but I got distracted by someone asking me if the PR was ready ;)

Will do it now 👍🏻